dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphUtility/BarGraph.cs

```diff
@@ -244,108 +266,108 @@ public void DrawText(WindowRenderTarget renderTarget, string titleText, string x
+            _textFormatHorizontal = new TextFormat(factoryDWrite, "Segoe", baseTextSize)
             {
                 TextAlignment = TextAlignment.Center,
                 ParagraphAlignment = ParagraphAlignment.Far
             };
-            textFormatVertical = new TextFormat(factoryDWrite, "Segoe", baseTextSize)
+            _textFormatVertical = new TextFormat(factoryDWrite, "Segoe", baseTextSize)
             {
                 TextAlignment = TextAlignment.Center,
                 ParagraphAlignment = ParagraphAlignment.Far
             };
 
             renderTarget.AntialiasMode = AntialiasMode.PerPrimitive;
-            renderTarget.TextAntialiasMode = TextAntialiasMode.Cleartype;
+            renderTarget.TextAntialiasMode = SharpDX.Direct2D1.TextAntialiasMode.Cleartype;
 
-            textSceneColorBrush = new SolidColorBrush(renderTarget, black);
+            _textSceneColorBrush = new SolidColorBrush(renderTarget, _black);
 
             var ClientRectangleTitle = new RectangleF(0, 0, textWidth, textHeight);
             var ClientRectangleXAxis = new RectangleF(0,
                 containerHeight - textHeight + sgOffsetY * 2, textWidth, textHeight);
             var ClientRectangleYAxis = new RectangleF(-sgOffsetX,
                 containerHeight - textHeight + sgOffsetY, textWidth, textHeight);
 
-            textSceneColorBrush.Color = black;
+            _textSceneColorBrush.Color = _black;
 
             //Draw title and x axis text.
             renderTarget.BeginDraw();
 
-            renderTarget.Clear(white);
-            renderTarget.DrawText(titleText, textFormatTitle, ClientRectangleTitle, textSceneColorBrush);
-            renderTarget.DrawText(xAxisText, textFormatHorizontal, ClientRectangleXAxis, textSceneColorBrush);
+            renderTarget.Clear(_white);
+            renderTarget.DrawText(titleText, _textFormatTitle, ClientRectangleTitle, _textSceneColorBrush);
+            renderTarget.DrawText(xAxisText, _textFormatHorizontal, ClientRectangleXAxis, _textSceneColorBrush);
 
             renderTarget.EndDraw();
 
             // Rotate render target to draw y axis text.
-            renderTarget.Transform = Matrix3x2.Rotation((float)(-Math.PI / 2), new Vector2(0, containerHeight));
+            renderTarget.Transform = SharpDX.Matrix3x2.Rotation((float)(-Math.PI / 2), new SharpDX.Vector2(0, containerHeight));
 
             renderTarget.BeginDraw();
 
-            renderTarget.DrawText(yAxisText, textFormatVertical, ClientRectangleYAxis, textSceneColorBrush);
+            renderTarget.DrawText(yAxisText, _textFormatVertical, ClientRectangleYAxis, _textSceneColorBrush);
 
             renderTarget.EndDraw();
 
             // Rotate the RenderTarget back.
-            renderTarget.Transform = Matrix3x2.Identity;
+            renderTarget.Transform = SharpDX.Matrix3x2.Identity;
         }
 
         //Dispose of resources.
         public void Dispose()
```

Comment:
This doesn't seem to get called. If it's important to dispose of the resources make sure it gets called. If it's not important then remove the dead code. I suspect it's important and what you need to do is make this class inherit IDisposable and make sure whatever class owns any IDisposable either inherits IDisposable too, or knows when to call Dispose.

Reply:
updated